### PR TITLE
Nominal ADC frequency and ppm correction

### PR DESCRIPTION
**Pull request by ik1xpv**
Hi,
This PR possibly solves issue #110. The parameter adcnominalfreq is defined. It is the nominal frequency value of the ADC, which is synthesised by the Si5351a PLL with a resolution approaching one Hertz.
Its value can be modified in the ExtIO.dll dialog between 50 and 140 MHz.
The gfFreqCorrectionPpm parameter represents the correction to be applied to the pll reference to correct its accuracy. Its value can be between +-200 ppm.
The ExtIO.dll dialogue is changed between the case where the dll is used by HDSDR that allows dynamically the choice of parameters like IF bandwidth from its GUI and the case of other SDR apps that do not offer this method.
The HF103 radio that uses a fixed frequency xtal is not controlled by this code at this moment. 
My proposed layout changes are as in the following picture.
image

The RF gain box controls HF attenuator and  LNA gain when the VHF tuner is used.
In HF the IF gain is used to control input VGA of RX888 mkII while is NA in RX888.
In VHF it controls the inner gains of the tuner.
Possibly the label IF may be changed in VGA (Voltage Gain Amplifier)  ?

Oscar

### CODE REVIEW

File: Core/config.h [hunk 1]
```diff
@@ -85,5 +86,8 @@ const uint32_t DEFAULT_ADC_FREQ = 64000000;	// ADC sampling frequency
 
 const uint32_t DEFAULT_TRANSFERS_PER_SEC = DEFAULT_ADC_FREQ / transferSamples;
 
+const uint32_t MIN_ADC_FREQ =  50000000;	   // ADC sampling frequency minimum
+const uint32_t MAX_ADC_FREQ = 140000000;	// ADC sampling frequency minimum
+const uint32_t N2_BANDSWITCH = 80000000;    // threshold 5 or 6 SR bandwidths
```
Comment (howard0su) [inline Core/config.h:91]: not sure about this const. what is this for?

File: ExtIO_sddc/ExtIO_sddc.cpp [hunk 2]
```diff
@@ -112,7 +112,19 @@ bool __declspec(dllexport) __stdcall InitHW(char *name, char *model, int& type)
 	if (!gbInitHW)
 	{
 		// do initialization
-		h_dialog = CreateDialog(hInst, MAKEINTRESOURCE(IDD_DLG_MAIN), NULL, (DLGPROC)&DlgMainFn);
+		// verify if HDSDR host name
+		SDR_progname[0] = 0;
```
Comment (howard0su) [inline ExtIO_sddc/ExtIO_sddc.cpp:116]: this is not needed as you declare this as BSS already.

**Review by howard0su — APPROVED** [review]
Looks good to me.

File: ExtIO_sddc/ExtIO_sddc.cpp (hunk 2 above)
Comment (ik1xpv) [reply ExtIO_sddc/ExtIO_sddc.cpp]: Thanks

File: Core/config.h (hunk 1 above)
Comment (ik1xpv) [reply Core/config.h]: I see no reason to keep it const. (I was lazy to write initialization code in config.cpp :-)

**Pull request merged**